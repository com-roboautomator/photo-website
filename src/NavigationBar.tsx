import { Button, createStyles, WithStyles } from "@material-ui/core";
import { withStyles } from "@material-ui/styles";
import React from "react";
import { RouteComponentProps } from "react-router-dom";
import Logo from "./images/Logo.png"

interface NavigationBarProps extends WithStyles<typeof styles>, React.PropsWithChildren<RouteComponentProps<any, any, unknown>> {
    selectedPage?: "Home" | "About" | "Contact" | "Gallery"
}

class NavigationBar extends React.Component<NavigationBarProps, any> {
    render() {
        const classes = this.props.classes
        const location = this.props.location.pathname

        return (
            <div className={classes.container}>
                    <div className={classes.logo}>
                        <img src={Logo} alt="" />
                    </div>
                <div className={classes.items_container}>
                    <div className={classes.wrapper}>
                        <div className={(location === "/") ? classes.line_selected : classes.line_unselected} >
                            <Button
                                className={(location === "/") ? classes.item_selected : classes.item_unselected}
                                href="/"
                            >
                                Home
                            </Button>
                        </div>
                    </div>
                    <div className={classes.wrapper}>
                        <div className={(location === "/gallery") ? classes.line_selected : classes.line_unselected}>
                            <Button
                                className={(location === "/gallery") ? classes.item_selected : classes.item_unselected}
                                href="/gallery"
                            >
                                Gallery
                            </Button>
                        </div>
                    </div>
                    <div className={classes.wrapper}>
                        <div className={(location === "/about") ? classes.line_selected : classes.line_unselected}>
                            <Button
                                className={(location === "/about") ? classes.item_selected : classes.item_unselected}
                                href="/about"
                            >
                                About
                            </Button>
                        </div>
                    </div>
                    <div className={classes.wrapper}>
                        <div className={(location === "/contact") ? classes.line_selected : classes.line_unselected}>
                            <Button
                                className={(location === "/contact") ? classes.item_selected : classes.item_unselected}
                                href="/contact"
                            >
                                Contact
                            </Button>
                        </div>
                    </div>
                </div>
            </div>
        );
    }
}

const styles = () => createStyles({
    container: {
        zIndex: 99,
        position: 'relative',
        display: 'flex',
        flex: 1,
        flexDirection: 'row',
        alignContent: 'center',
        justifyContent: 'space-between',
        boxShadow: '1px 2px 10px rgba(0, 0, 0, 0.6)',
        paddingRight: 80,
    },
    items_container: {
        display: 'flex',
        flexDirection: 'row',
    },

    logo: {
        paddingLeft: '70px',
    },
    item_selected: {
        fontSize: 16,
        fontfamily: "Open Sans",
        alignSelf: 'center',
        color: 'black',
    },
    item_unselected: {
        fontSize: 16,
        fontfamily: "Open Sans",
        alignSelf: 'center',
        color: 'gray',
    },
    wrapper: {
        flexDirection: 'row',
        justifyContent: 'center',
        alignItems: 'center',
        paddingLeft: 12,
        paddingRight: 12,
        paddingTop: 16,
        paddingBottom: 8,
    },
    line_selected: {
        paddingBottom: 0,
        borderBottomStyle: 'solid',
        borderBottomWidth: 4,
        borderBottomColor: '#7baab5',
    },
    line_unselected: {
    }
})

export default withStyles(styles)(NavigationBar)