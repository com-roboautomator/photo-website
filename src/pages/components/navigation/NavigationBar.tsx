import {createStyles, WithStyles} from '@material-ui/core'
import {withStyles} from '@material-ui/styles'
import React from 'react'
import {RouteComponentProps} from 'react-router-dom'
import Button from './NavigationMenuButton'
import Logo from './NavigationMenuLogo'

interface NavigationBarProps
    extends WithStyles<typeof styles>,
        React.PropsWithChildren<RouteComponentProps<any, any, unknown>> {}

class NavigationBar extends React.Component<NavigationBarProps, any> {
    render() {
        const classes = this.props.classes
        const location = this.props.location.pathname

        return (
            <div className={classes.container} data-testid={'NavigationBar'}>
                <Logo />
                <div className={classes.items_container}>
                    <Button text="HOME" path="/" selected={location === '/'} />
                    <Button
                        text="GALLERY"
                        path="/gallery"
                        selected={location.includes('/gallery')}
                    />
                    <Button
                        text="ABOUT"
                        path="/about"
                        selected={location === '/about'}
                    />
                    <Button
                        text="CONTACT"
                        path="/contact"
                        selected={
                            location === '/contact' ||
                            location === '/contact/confirmation'
                        }
                    />
                </div>
            </div>
        )
    }
}

const styles = () =>
    createStyles({
        container: {
            zIndex: 99,
            //position: 'relative',
            display: 'flex',
            flex: 1,
            flexDirection: 'row',
            alignContent: 'center',
            justifyContent: 'space-between',
            boxShadow: '1px 2px 10px rgba(0, 0, 0, 0.9)',
            paddingRight: 80,
        },
        items_container: {
            display: 'flex',
            flexDirection: 'row',
        },
    })

export default withStyles(styles)(NavigationBar)
