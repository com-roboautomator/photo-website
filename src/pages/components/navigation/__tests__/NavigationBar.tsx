import NavigationBar from '../NavigationBar'
import React from 'react'
import {BrowserRouter, Route} from 'react-router-dom'
import {render} from '@testing-library/react'
import renderer from 'react-test-renderer'

describe('NavigationBar', () => {
    it('renders correctly when there are no items', () => {
        const navigationBarRender = renderer
            .create(
                <BrowserRouter>
                    <Route
                        render={(props: any) => <NavigationBar {...props} />}
                    />
                </BrowserRouter>
            )
            .toJSON()

        expect(navigationBarRender).toMatchSnapshot()
    })

    it('should render NavigationBar', () => {
        // When
        // Given
        const component = render(
            <BrowserRouter>
                <Route render={(props: any) => <NavigationBar {...props} />} />
            </BrowserRouter>
        )

        // Then
        expect(component.queryByTestId('NavigationBar')).toBeTruthy()
    })

    it('should render Home button', () => {
        // When
        // Given
        const component = render(
            <BrowserRouter>
                <Route render={(props: any) => <NavigationBar {...props} />} />
            </BrowserRouter>
        )

        // Then
        expect(component.queryByTestId('Button-HOME')).toBeTruthy()
        expect(
            component.getByTestId('Button-HOME-Button').firstChild?.textContent
        ).toEqual('HOME')
    })

    it('should render Gallery button', () => {
        // When
        // Given
        const component = render(
            <BrowserRouter>
                <Route render={(props: any) => <NavigationBar {...props} />} />
            </BrowserRouter>
        )

        // Then
        expect(component.queryByTestId('Button-GALLERY')).toBeTruthy()
        expect(
            component.getByTestId('Button-GALLERY-Button').firstChild
                ?.textContent
        ).toEqual('GALLERY')
    })

    it('should render About button', () => {
        // When
        // Given
        const component = render(
            <BrowserRouter>
                <Route render={(props: any) => <NavigationBar {...props} />} />
            </BrowserRouter>
        )

        // Then
        expect(component.queryByTestId('Button-ABOUT')).toBeTruthy()
        expect(
            component.getByTestId('Button-ABOUT-Button').firstChild?.textContent
        ).toEqual('ABOUT')
    })

    it('should render Contact button', () => {
        // When
        // Given
        const component = render(
            <BrowserRouter>
                <Route render={(props: any) => <NavigationBar {...props} />} />
            </BrowserRouter>
        )

        // Then
        expect(component.queryByTestId('Button-CONTACT')).toBeTruthy()
        expect(
            component.getByTestId('Button-CONTACT-Button').firstChild
                ?.textContent
        ).toEqual('CONTACT')
    })
})
