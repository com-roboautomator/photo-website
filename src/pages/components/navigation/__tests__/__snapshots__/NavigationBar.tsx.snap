// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationBar renders correctly when there are no items 1`] = `
<div
  className="NavigationBar-container-1"
  data-testid="NavigationBar"
>
  <div
    className="NavigationMenuLogo-logo-3"
    data-testid="Menu-Logo"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <img
        alt=""
        data-testid="Menu-Logo-Image"
        src="Logo.png"
      />
    </a>
  </div>
  <div
    className="NavigationBar-items_container-2"
  >
    <div
      className="NavigationMenuButton-wrapper-4"
      data-testid="Button-HOME"
    >
      <div
        className="NavigationMenuButton-line_selected-7"
        data-testid="Button-HOME-Underline"
      >
        <a
          aria-disabled={false}
          className="MuiButtonBase-root MuiButton-root MuiButton-text NavigationMenuButton-item_selected-5"
          data-testid="Button-HOME-Button"
          href="/"
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
        >
          <span
            className="MuiButton-label"
          >
            HOME
          </span>
        </a>
      </div>
    </div>
    <div
      className="NavigationMenuButton-wrapper-4"
      data-testid="Button-GALLERY"
    >
      <div
        className="NavigationMenuButton-line_unselected-8"
        data-testid="Button-GALLERY-Underline"
      >
        <a
          aria-disabled={false}
          className="MuiButtonBase-root MuiButton-root MuiButton-text NavigationMenuButton-item_unselected-6"
          data-testid="Button-GALLERY-Button"
          href="/gallery"
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
        >
          <span
            className="MuiButton-label"
          >
            GALLERY
          </span>
        </a>
      </div>
    </div>
    <div
      className="NavigationMenuButton-wrapper-4"
      data-testid="Button-ABOUT"
    >
      <div
        className="NavigationMenuButton-line_unselected-8"
        data-testid="Button-ABOUT-Underline"
      >
        <a
          aria-disabled={false}
          className="MuiButtonBase-root MuiButton-root MuiButton-text NavigationMenuButton-item_unselected-6"
          data-testid="Button-ABOUT-Button"
          href="/about"
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
        >
          <span
            className="MuiButton-label"
          >
            ABOUT
          </span>
        </a>
      </div>
    </div>
    <div
      className="NavigationMenuButton-wrapper-4"
      data-testid="Button-CONTACT"
    >
      <div
        className="NavigationMenuButton-line_unselected-8"
        data-testid="Button-CONTACT-Underline"
      >
        <a
          aria-disabled={false}
          className="MuiButtonBase-root MuiButton-root MuiButton-text NavigationMenuButton-item_unselected-6"
          data-testid="Button-CONTACT-Button"
          href="/contact"
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
        >
          <span
            className="MuiButton-label"
          >
            CONTACT
          </span>
        </a>
      </div>
    </div>
  </div>
</div>
`;
